node/src/command/solochain.rs

```diff
@@ -0,0 +1,415 @@
+        self,
+        initialize: impl FnOnce(SolochainConfig) -> F,
+    ) -> std::result::Result<(), E>
+    where
+        F: Future<Output = std::result::Result<TaskManager, E>>,
+        E: std::error::Error + Send + Sync + 'static + From<sc_service::Error>,
+    {
+        self.print_node_infos();
+
+        let mut task_manager = self.tokio_runtime.block_on(initialize(self.config))?;
+
+        let res = self
+            .tokio_runtime
+            .block_on(self.signals.run_until_signal(task_manager.future().fuse()));
+        // We need to drop the task manager here to inform all tasks that they should shut down.
+        //
+        // This is important to be done before we instruct the tokio runtime to shutdown. Otherwise
+        // the tokio runtime will wait the full 60 seconds for all tasks to stop.
+        let task_registry = task_manager.into_task_registry();
+
+        // Give all futures 60 seconds to shutdown, before tokio "leaks" them.
+        let shutdown_timeout = Duration::from_secs(60);
+        self.tokio_runtime.shutdown_timeout(shutdown_timeout);
+
+        let running_tasks = task_registry.running_tasks();
+
+        if !running_tasks.is_empty() {
+            log::error!("Detected running(potentially stalled) tasks on shutdown:");
+            running_tasks.iter().for_each(|(task, count)| {
+                let instances_desc = if *count > 1 {
+                    format!("with {} instances ", count)
+                } else {
+                    "".to_string()
+                };
+
+                if task.is_default_group() {
+                    log::error!(
+                        "Task \"{}\" was still running {}after waiting {} seconds to finish.",
+                        task.name,
+                        instances_desc,
+                        shutdown_timeout.as_secs(),
+                    );
+                } else {
+                    log::error!(
+						"Task \"{}\" (Group: {}) was still running {}after waiting {} seconds to finish.",
+						task.name,
+						task.group,
+						instances_desc,
+						shutdown_timeout.as_secs(),
+					);
+                }
+            });
+        }
+
+        res.map_err(Into::into)
+    }
+}
+
+/// Equivalent to [Cli::create_runner]
+pub fn create_runner<T: CliConfiguration<DVC>, DVC: DefaultConfigurationValues>(
+    command: &T,
+) -> sc_cli::Result<SolochainRunner> {
+    let tokio_runtime = sc_cli::build_runtime()?;
+
+    // `capture` needs to be called in a tokio context.
+    // Also capture them as early as possible.
+    let signals = tokio_runtime.block_on(async { Signals::capture() })?;
+
+    init_cmd(command, &Cli::support_url(), &Cli::impl_version())?;
+
+    let base_path = command.base_path()?.unwrap();
+    let network_node_name = command.node_name()?;
+    let is_dev = command.is_dev()?;
+    let role = command.role(is_dev)?;
+    // TODO: where to get relay_chain_id from?
+    let relay_chain_id = "starlight_local_testnet".to_string();
```

Comment:
from the specs I think you can take the relay chain id

Reply:
It's complicated, see the other discussion.

*disregard*